### PR TITLE
update libraries

### LINKED ISSUE #149: Update Contract dependecies and fix Audit results
Update libraries and tools
- Zos 2.2.3
- truffle v5.0.17
- ganache cli - might have an impact on gas cost estimations
- truffle hd provider

Run npm audit and update/fix dependencies based on result

### PR DESCRIPTION
**Pull request by rdinicut**
closes #149

### CODE REVIEW

File: test/Gas.test.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ contract("Gas costs", function (accounts) {
     describe("Check the gas cost for preCommit and commit with the identity proxy for ACTION key", async function () {
 
         const preCommitMaxGas = 95000;
-        const commitMaxGas = 85000;
+        const commitMaxGas = 175000;
```
Comment (rdinicut) [inline test/Gas.test.js]: ganache improved gas estimates. Actual is '174100'

**Review by pstehlik — APPROVED** [review]
LGTM - i have not checked all the newest version numbers of the libs though but trust that you updated to the most relevant

File: test/Gas.test.js (hunk 1 above)
Comment (xmxanuel) [reply test/Gas.test.js]: Maybe we can add a variable `buffer`?
`const commitMaxGas = 174100 + buffer;`?

Then it is clear for everybody.

**Review by xmxanuel — APPROVED** [review]
LGTM

**Pull request merged**